src/components/GuestBook/GuestBook.jsx

```diff
@@ -0,0 +1,79 @@
+import { useState } from 'react';
+import { useEntries } from '../../context/EntryContext';
+import { useUser } from '../../context/UserContext';
+
+export default function GuestBook() {
+  const [name, setName] = useState('');
+  const [guestEntry, setGuestEntry] = useState('');
+  const { entries, setEntries } = useEntries();
+  const { user, setUser } = useUser();
+
+  function updateGuestName() {
+    if (!guestEntry) return setUser(name);
+    setEntries([...entries, { name, message: guestEntry }]);
+    setGuestEntry('');
+  }
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    updateGuestName();
+  };
+
+  const guestNameInput = (
+    <div>
+      <div>
+        <label className="labelText">Guest Name</label>
+      </div>
+      <div>
+        <input
+          id="guestName"
+          type="text"
+          placeholder="Guest Name..."
+          value={name}
+          onChange={(event) => setName(event.target.value)}
+        />
+      </div>
+    </div>
+  );
+
+  const displayMessage = user
+    ? `Thanks for signing the guestbook ${user}!`
+    : 'Please sign the guestbook!';
+
+  return (
+    <>
+      <h1>{displayMessage}</h1>
+      <form onSubmit={handleSubmit}>
+        {user ? null : guestNameInput}
+        <div>
+          <div>
+            <label className="labeltext">Guest Entry</label>
+          </div>
+          <div>
+            <textarea
+              id="guestEntry"
+              value={guestEntry}
+              placeholder="Your entry!"
+              onChange={(event) => setGuestEntry(event.target.value)}
+            />
+          </div>
+        </div>
+        <div>
+          <button className="button" type="submit">
+            Click to Sign!
+          </button>
+          {user && (
+            <button
```

Comment:
this doesn't appear to be rendering to the page, either. Since the header isn't updating with the user name, and this isn't rendering EITHER, it leads me to suspect that perhaps your user object is not getting updated properly. 

Give that a look over and let me know if you have any questions about it!

Reply:
Hey Juli! I pushed up our fixes so useUser() is now functioning as it should. Could you review the changes and adjust my grade in Canvas? Thank you!